### PR TITLE
[1387] Add a rolled_over state for users

### PR DESCRIPTION
**Pull request by fofr**
### Context

- Use user state to determine if they've seen the rollover screen
- They must have seen the transition screen first

Goes with https://github.com/DFE-Digital/manage-courses-frontend/pull/430

### CODE REVIEW

**Comment by fofr** [comment]
[[S] Rollover Screen - first log-in for users when next cycle is available](https://trello.com/c/CqD9fZa4/1387-s-rollover-screen-first-log-in-for-users-when-next-cycle-is-available)

File: app/models/user.rb [hunk 1]
```diff
@@ -38,10 +38,15 @@ class User < ApplicationRecord
   aasm column: 'state' do
     state :new, initial: true
     state :transitioned
+    state :rolled_over
```
Comment (misaka) [inline app/models/user.rb]: Is it premature to use `rolled_over_2020` here?

Comment (fofr) [reply app/models/user.rb]: Next year we can always do `rolled_over_2021` if we need to. Not sure it's needed yet.

**Comment by fofr** [comment]
[[M] Rollover screen - record first log-in for users when next cycle is available](https://trello.com/c/Qyn9ZuKT/1545-m-rollover-screen-record-first-log-in-for-users-when-next-cycle-is-available)

**Pull request merged**